File: bubblesort.cpp
#include <iostream>
#include <bits/stdc++.h>"
using namespace std;

void swap(int *xp, int *yp) //Swaps the value of the pointers for use in the bubble sort "for" loop
{
	int tmp = *xp;
	*xp = *yp;
	*yp = tmp;
}

void printArr(int arr[], int size) //Simple loop to iterate through the array and print it
{
	for (int i = 0; i < size; i++)
	{
		cout << arr[i] << " ";
		cout << endl;
	}
}

void bubbleSort(int arr[], int n)
{
	int i, j;
	for (i = 0; i < n-1; i++)
	{
		for (j = 0; j < n-i-1; j++)
		{
			if (arr[j] > arr[j+1])
			{
				swap(&arr[j], &arr[j+1]);
			}
		}
	}
}


int main()
{
    //Test case arrays
	int test1[] = {23, 78, 43, 14, 90, 82};
	int test2[] = {65, 32, 17, 75, 55, 4};

	int n = sizeof(test1)/sizeof(test1[0]);
	int o = sizeof(test2)/sizeof(test2[0]);

	bubbleSort(test1, n);
	bubbleSort(test2, o);

	cout << "Test array 1: \n";
	printArr(test1, n);
	cout << "Test array 2: \n";
	printArr(test2, o);
	return 0;
}

File: bubblesort.js
function swap(arr, x, y) //Takes the target array and the indexes of the desired numbers to swap
{
    var tmp, first = x, second = y, xVal = arr[x], yVal = arr[y];
    tmp = second;
    second = first;
    first = tmp;
    arr.splice(x, 1, yVal);
    arr.splice(y, 1, xVal);
}

function bubblesort(arr)
{
    for (let i = 0; i < arr.length; i++)
    {
        for (let j = 0; j < arr.length; j++)
        {
            if (arr[j] > arr[j + 1])
            {
                swap(arr, j, j + 1);
            }
        }
    }
}

//Test cases
var test1 = [23, 78, 43, 14, 90, 82];
var test2 = [65, 32, 17, 75, 55, 4];
bubblesort(test1);
bubblesort(test2);
console.log("Test 1 sort: " + test1);
console.log("Test 2 sort: " + test2);